packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -0,0 +1,3369 @@
+              dappSuggestedGasFees: undefined,
+              deviceConfirmedOn: undefined,
+              id: 'c4cc0ff0-ba28-11ee-926f-55a7f9c2c2c6',
+              origin: undefined,
+              securityAlertResponse: undefined,
+              status: 'approved',
+              time: 1706039113766,
+              txParams: {
+                from: '0x6bf137f335ea1b8f193b8f6ea92561a60d23a207',
+                gas: '0x5208',
+                nonce: '0x1',
+                to: '0x08f137f335ea1b8f193b8f6ea92561a60d23a211',
+                value: '0x0',
+                maxFeePerGas: '0x1',
+                maxPriorityFeePerGas: '0x1',
+              },
+              userEditedGasLimit: false,
+              verifiedOnBlockchain: false,
+              type: 'simpleSend',
+              networkClientId: 'sepolia',
+              simulationFails: undefined,
+              originalGasEstimate: '0x5208',
+              defaultGasEstimates: {
+                gas: '0x5208',
+                maxFeePerGas: '0x1',
+                maxPriorityFeePerGas: '0x1',
+                gasPrice: undefined,
+                estimateType: 'dappSuggested',
+              },
+              userFeeLevel: 'dappSuggested',
+              sendFlowHistory: [],
+              history: [
+                {
+                  chainId: '0xaa36a7',
+                  id: 'c4cc0ff0-ba28-11ee-926f-55a7f9c2c2c6',
+                  status: 'unapproved',
+                  time: 1706039113766,
+                  txParams: {
+                    from: '0x6bf137f335ea1b8f193b8f6ea92561a60d23a207',
+                    to: '0x08f137f335ea1b8f193b8f6ea92561a60d23a211',
+                    value: '0x0',
+                    gas: '0x5208',
+                    maxFeePerGas: '0x1',
+                    maxPriorityFeePerGas: '0x1',
+                  },
+                  userEditedGasLimit: false,
+                  verifiedOnBlockchain: false,
+                  type: 'simpleSend',
+                  networkClientId: 'sepolia',
+                  originalGasEstimate: '0x5208',
+                  defaultGasEstimates: {
+                    gas: '0x5208',
+                    maxFeePerGas: '0x1',
+                    maxPriorityFeePerGas: '0x1',
+                    estimateType: 'dappSuggested',
+                  },
+                  userFeeLevel: 'dappSuggested',
+                  sendFlowHistory: [],
+                },
+                [
+                  {
+                    op: 'add',
+                    path: '/txParams/nonce',
+                    value: '0x1',
+                    note: 'TransactionController#approveTransaction - Transaction approved',
+                    timestamp: 1706039113767,
+                  },
+                  {
+                    op: 'replace',
+                    path: '/status',
+                    value: 'approved',
+                  },
+                ],
+              ],
+            },
+          ],
+        },
+      });
```

Comment:
can probably trim these mock transactions down quite a bit. Dropping all of the elements in `history` is probably okay

Reply:
working on this

Reply:
https://github.com/MetaMask/core/pull/3643/commits/4e9d346cbbb05e49e4248c47a5efcebc8253c2ad